index.js

```diff
@@ -192,6 +199,19 @@ module.exports = function configParser({
     // Convert from YAML to JSON
     return (
         parseYaml(yaml)
+            // Merge Pipeline template
+            .then(parsedDoc =>
+                phaseMerge(
+                    parsedDoc,
+                    pipelineTemplateVersionFactory,
+                    pipelineTemplateTagFactory,
+                    pipelineTemplateFactory
+                ).then(({ warnings, mergedDoc }) => {
+                    warnMessages = warnMessages.concat(warnings);
+
+                    return mergedDoc;
+                })
```

Comment:
structural validation is not data-schema validation, data-schema validation happens before merge at `parseYaml`

Reply:
added another check right before merging

Reply:
Let us move the logic to validate out of `merge.js`. It will be good not to overload a module to handle multiple things (validating, merging, etc).

We can use the existing `structural.js` module to perform validation before/after merging. We can modify it to take a boolean flag `skipAdditionalRules`.